lib/excon/connection.rb

```diff
@@ -358,6 +359,12 @@ def validate_params(validation, params)
         #params = params.dup
         #invalid_keys.each {|key| params.delete(key) }
       end
+
+      if validation == :connection && params.key?(:host) && !params.key?(:hostname)
+        Excon.display_warning('hostname is missing! For IPv6 support, provide both host and hostname: Excon::Connection#new(:host => uri.host, :hostname => uri.hostname, ...).')
```

Comment:
In my experience other stuff usually uses `Excon.new` (I think). It is probably ok and I guess if it isn't we'll probably hear about it pretty quickly (and accidentally warning is not the end of the world).